internal/worker/uniter/runner/runner.go

```diff
@@ -316,6 +319,22 @@ func (runner *runner) runJujuExecAction() (err error) {
 	return runner.context.Flush("juju-exec", err)
 }
 
+// scopedActionCancel returns a context that is cancelled when either the
+// supplied context is cancelled or the supplied abort channel is closed.
+// This is only required until actions are refactored to use a context.
+func scopedActionCancel(ctx stdcontext.Context, abort <-chan struct{}) stdcontext.Context {
+	c, cancel := stdcontext.WithCancel(ctx)
+	go func() {
+		defer cancel()
+
+		select {
+		case <-ctx.Done():
+		case <-abort:
+		}
+	}()
+	return c
+}
```

Comment:
The consequence of mixing action abort with context. I wonder if there is a better pattern to handle this without storing a context on the action itself.